$DragonFly: src/gnu/usr.bin/cvs/cvs/Attic/parseinfo.c.patch,v 1.2 2005/03/13 11:43:07 corecode Exp $
--- /usr/src/contrib/cvs-1.12.11/src/parseinfo.c	2005-01-05 22:17:43.000000000 +0100
+++ parseinfo.c	2005-03-06 23:57:22.000000000 +0100
@@ -333,8 +333,8 @@
  *
  *   xmalloc() failures are fatal, per usual.
  */
-struct config *
-parse_config (const char *cvsroot)
+static struct config *
+parse_config_file (const char *cvsroot, const char *config_name, struct config *retval)
 {
     char *infopath;
     FILE *fp_info;
@@ -342,13 +342,13 @@
     size_t line_allocated = 0;
     size_t len;
     char *p;
-    struct config *retval;
 
     TRACE (TRACE_FUNCTION, "parse_config (%s)", cvsroot);
 
-    retval = new_config ();
+    if (retval == NULL)
+	retval = new_config ();
 
-    infopath = Xasprintf ("%s/%s/%s", cvsroot, CVSROOTADM, CVSROOTADM_CONFIG);
+    infopath = Xasprintf ("%s/%s/%s", cvsroot, CVSROOTADM, config_name);
 
     fp_info = CVS_FOPEN (infopath, "r");
     if (fp_info == NULL)
@@ -431,9 +431,11 @@
 	    readBool (infopath, "SystemAuth", p, &dummy);
 	}
 #endif
-	else if (strcmp (line, "LocalKeyword") == 0)
+	else if (strcmp (line, "LocalKeyword") == 0 ||
+		 strcmp (line, "tag") == 0)
 	    RCS_setlocalid (&retval->keywords, p);
-	else if (strcmp (line, "KeywordExpand") == 0)
+	else if (strcmp (line, "KeywordExpand") == 0 ||
+		 strcmp (line, "tagexpand") == 0)
 	    RCS_setincexc (&retval->keywords, p);
 	else if (strcmp (line, "PreservePermissions") == 0)
 #ifdef PRESERVE_PERMISSIONS_SUPPORT
@@ -547,3 +549,15 @@
 	free (line);
     return retval;
 }
+
+struct config *
+parse_config (const char *cvsroot)
+{
+    struct config *retval;
+
+    retval = NULL;
+    retval = parse_config_file (cvsroot, CVSROOTADM_CONFIG, retval);
+    retval = parse_config_file (cvsroot, CVSROOTADM_OPTIONS, retval);
+
+    return retval;
+}
